### ISSUE TITLE
[Bug] Investigate obstacles for latest CRIS data not being ingested

### ISSUE THREAD

**Issue by patrickm02L** [cityofaustin/atd-data-tech#7937]
On Wednesday, December 8th, Xavier A. notified the DTS team that there were problems with the latest CRIS data being ingested in VZDB.

@frankhereford and @mateoclarke looked into this and discovered the following:

> We traced the flow of data from the TXDOT upload, through the SFTP endpoint, to the AWS S3 bucket, and into Airflow. We observed airflow to not be running, which Mateo linked to server software upgrades which happened yesterday, but not entirely successfully. 
>
>This was reported to the team, and Sergio stepped in and has begun the process of restoring Airflow.
>
> Once airflow is up and running, we’ll initiate the DAG to process the waiting data from TxDOT and confirm that it makes it through to the VZDB.

Restoration of the Airflow allowed crashes to be imported from the CRIS CSV records and the PDF CR3s for those same crashes were downloaded.

**Comment by patrickm02L** [comment]
@frankhereford Please add any details about any further obstacles that are affecting the ingestion.

**Comment by frankhereford** [comment]
This morning when I went to run the CR3 PDF downloading software, I observed there were no new crashes. Because we did an import yesterday afternoon, this wasn't entirely surprising. I looked on the SFTP endpoint where TxDOT uploads the daily extract of crash data, and I did not see a zip file or CSV files waiting to be processed, nor did I see any new files located in the S3 bucket. 

At this point, I began to suspect there may be a continuing interruption in our ingest pipeline, but I have not seen a specific error to confirm this. It would be tomorrow morning, if there is another zero crashes to import, that I would become fully convinced there is something not operating as intended. 

I conversed with @patrickm02L and conveyed the above information, and he kindly created this ticket to both document the work spent on this issue over the last few days on this issue and to coordinate any further action, if needed.

**Comment by patrickm02L** [comment]
Whoa, that drops off is suss! No wonder you were alerting us that there may be more to the CRIS ingestion problem than what happened yesterday. 

Yes, I think this query might be useful when we get around to building daily reports in the VZE. Ref #5973, #6989

**Comment by frankhereford** [comment]
I have run the crashes-by-day query again this morning, and it appears we have not taken in any new crashes. I'll be investigating and working this issue this morning, per instructions from @patrickm02L. Thanks!

**Comment by patrickm02L** [comment]
@frankhereford Thanks for being proactive and diagnosing the problem early. Please go ahead and tackle this.

**Comment by frankhereford** [comment]
I'll know for sure tomorrow AM (and I'll check and report back here), but I think we're back on the rails. 

For this particular case, the cause of our pause in crash intake was due to our recurring CRIS extract having an end-date of the Dec-08. Therefore, CRIS was not sending us newly processed crashes. I have created a new extract (with a label prepended with FLH) that will continue to deliver us through the end of the year. It uses the password coded into the secrets file used by the cron task to unzip the data.

I observed the delivery of the extract onto the SFTP endpoint, and then manually ran the cron job which unzips and uploads I have manually triggered the pair of DAGs to process the data. The above query was used to quantify the new crash temporal distribution. Here's the graph for those who are interested:

![image](https://user-images.githubusercontent.com/3653206/145605736-621b5881-16b7-40b3-9589-d7628cde72eb.png)

The processing-lag tail is there, but again, that is not indicating a problem -- just that crashes come in between 3-10 days after they are occur, and sometimes even later.

I have downloaded CR3 PDFs for the new imports, and that should give the VZ team the latest and greatest data, in as much as it is available to us. I'll close this issue tomorrow when I observe new crashes' tomorrow AM showing that the automated process is operational.
